### PR TITLE
JDK-8329692 : Add more details to FrameStateTest.java test instructions

### PR DESCRIPTION
**Pull request by honkar-jdk**
For the following manual test, more instructions are added as to what to expect for "hide,iconify and show" vs "hide,iconify,show and restore" for clarity.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8329692](https://bugs.openjdk.org/browse/JDK-8329692): Add more details to FrameStateTest.java test instructions (**Bug** - P4)

### Reviewers
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - Committer) ⚠️ Review applies to [88f5d048](https://git.openjdk.org/jdk/pull/19008/files/88f5d048e9177812ca36b9ef8fb8f10ed6072170)
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**) ⚠️ Review applies to [88f5d048](https://git.openjdk.org/jdk/pull/19008/files/88f5d048e9177812ca36b9ef8fb8f10ed6072170)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19008/head:pull/19008` \
`$ git checkout pull/19008`

Update a local copy of the PR: \
`$ git checkout pull/19008` \
`$ git pull https://git.openjdk.org/jdk.git pull/19008/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19008`

View PR using the GUI difftool: \
`$ git pr show -t 19008`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19008.diff">https://git.openjdk.org/jdk/pull/19008.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19008#issuecomment-2083856991)

### CODE REVIEW

File: test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java [hunk 1]
```diff
@@ -74,23 +74,26 @@ This test checks that when setState(Frame.ICONIFIED) is called before
...
-        (Frame shown in proper state).<br>
+        up make sure the proper behavior occurred (Frame shown in proper state).<br>
         Click the Dispose button to close the Frame.<br>
 
         </p><hr/><p>
 
         Do the above steps for all the different Frame state combinations available.<br>
-        If you observe the proper behavior the test has passed, Press the Pass button.<br>
-        Otherwise the test has failed, Press the Fail button.
+        For "hide,iconify and show" case, the frame is hidden then iconified hence Window2
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java:91]: Minor: 
```suggestion
        For "hide, iconify and show" case, the frame is hidden then iconified hence Window2
```

File: test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java [hunk 2]
```diff
@@ -74,23 +74,26 @@ This test checks that when setState(Frame.ICONIFIED) is called before
...
+        is not seen on-screen when shown as the frame is still in the ICONIFIED state.
+        Window2 is visible on-screen when it is restored to NORMAL state as observed with
+        "hide,iconify,show and restore" case.<br><br>
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java:94]:
```suggestion
        "hide, iconify, show and restore" case.<br><br>
```

File: test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java [hunk 3]
```diff
@@ -320,11 +323,11 @@ public void windowOpened(WindowEvent e) {
 
         public void stateLog(String message) {
             PassFailJFrame
-                .log("[State=%d] %s %s".formatted(getState(), name, message));
+                .log("[Current State=%d] %s %s".formatted(getState(), name, message));
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java:326]:
```suggestion
                .log("[Current State = %d] %s %s".formatted(getState(), name, message));
```

File: test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java [hunk 4]
```diff
@@ -320,11 +323,11 @@ public void windowOpened(WindowEvent e) {
...
         }
 
         public void stateLog() {
-            PassFailJFrame.log("[State=" + getState() + "]");
+            PassFailJFrame.log("[Current State=" + getState() + "]");
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java:330]:
```suggestion
            PassFailJFrame.log("[Current State = " + getState() + "]");
```

File: test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java [hunk 5]
```diff
@@ -119,7 +122,7 @@ public static void main(String[] args) throws Exception {
                 .title("GetBoundsResizeTest Instructions")
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java:122]: Looks like title is mismatched with the test. Should it be "Frame State and Size Test Instructions" ?

**Review by kumarabhi006 — COMMENTED** [review]
Since these lines are unchanged, can't add comments there.

1. AT L155, space after `=` is missing.
`icontst =new CreateFrame(cbIconState.getState(), cbResize.getState());`

2. In `CreateFrame` constructor, calling `pack()` is redundant as frame size is set by `setBounds(...)` method.

3. L24 to L31 may be removed. Consolidated summary can be defined in `jtreg summary tag`.

**Comment by honkar-jdk** [comment]
> One observation is that, after the updates the size of whole test window is slightly increased in height where the second Frame which has options is out of the window frame. When the taskbar is at bottom nothing seems to be visible.
> One option is to limit the instruction rows or to use `PassFailJFrame.splitUI` instead of `PassFailJFrame.splitUIBottom` (Offcourse in splitUI the second Frame occupies the whole screen length which is not required).

PassFailJFrame.splitUI is not a good option as it causes a lot of whitespace in this case. I have reduced the log area instead.

File: test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java (hunk 5 above)
Comment (honkar-jdk) [reply test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java]: Updated.

File: test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java [hunk 6]
```diff
@@ -106,22 +101,22 @@ After the Frame (Frame State Test (Window2)) comes
...
+    Checkbox cbNonResize = new Checkbox("Frame Non-Resizable", cbgResize, false);
     Checkbox cbResize = new Checkbox("Frame Resizable", cbgResize, true);
 
     CreateFrame icontst;
 
     public static void main(String[] args) throws Exception {
         PassFailJFrame
                 .builder()
-                .title("GetBoundsResizeTest Instructions")
+                .title("Frame Stateand Size Test Instructions")
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java:114]:
```suggestion
                .title("Frame State and Size Test Instructions")
```

File: test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java [hunk 7]
```diff
@@ -175,13 +170,13 @@ static class CreateFrame extends Frame
...
-            add(b3 = new Button("iconify"));
-            add(b4 = new Button("iconify and restore"));
-            add(b5 = new Button("hide and show"));
-            add(b6 = new Button("hide, iconify and show"));
-            add(b7 = new Button("hide, iconify, show, and restore"));
+            add(b1 = new Button("Resizable"));
+            add(b2 = new Button("Resize"));
+            add(b3 = new Button("Iconify"));
+            add(b4 = new Button("Iconify and Restore"));
+            add(b5 = new Button("Hide and show"));
```
Comment (lawrence-andrew) [inline test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java:177]: To be consistent with other buttons , this button should can be changed from  text "Hide and show"   to "Hide and show"

**Review by TejeshR13 — APPROVED** [review]

**Review by azvegint — APPROVED** [review]

**Pull request closed without merging**